### ISSUE TITLE
GOB parser and encoder

### ISSUE THREAD

**Issue by KristianLyng** [telenornms/skogul#227]
While json-encoded HTTPS is versatile, it is not particularly efficient. Protocol buffers are not self describing, so they are not well suited for inter-skogul communication either. 

However, gobs ( https://pkg.go.dev/encoding/gob - https://go.dev/blog/gob ) should be the right fit for skogul-to-skogul traffic. It can be transported with any of the existing senders.

Gobs are only selfdescribing within the same stream, so a new stream will have to be created for each container, this reduces the effectiveness somewhat, but is required to avoid two-way communication between encoder and decoder to determine when to re-send the headers/"reboot" the encoder after a decoder restart/loss of state. 

In short: Whenever the encoder is called, a new bytes.Buffer and gob.NewEncoder() needs to be set up. We should probably add benchmarks to both this and the json encoder to compare. Regardless of performance, I believe it's worthwhile because with JSON we have information loss due to lack of typing: We can't determine if '0' is an integer or a floating point. 

EncodeMetric() and Encode() can be identical.

**Comment by KristianLyng** [comment]
Switch "Sender" with "encoder" and we're there, this should actually be pretty darn quick now.
